<?xml version="1.0"?>
<launch>
    <!-- <master auto="start"/>  -->
    <!--   <param name="use_sim_time" value="true" /> -->

    <!--     param file -->
    <arg name="param_yaml_file" value="$(find patrol_robot)/param/fws_real.yaml" />
    <param name="param_yaml_file" value="$(arg param_yaml_file)" />

    <!--   robot model -->
    <node pkg="patrol_robot" type="fws_model.py" name="fws_model"/>

    <!--     tf_manager -->
    <node pkg="patrol_robot" type="tf_manager.py" name="tf_manager"/>

    <!--   map_server -->
    <node pkg="map_server" type="map_server" name="map_server" args="$(find patrol_robot)/map/fws_map_sim.yaml"/>

    <!--   if sim, process vlp16 in sim -->
    <include file="$(find patrol_robot)/launch/fws_handle_sim_vlp16.launch" />

    <!--      als -->
    <include file="$(find als_ros)/launch/mcl_patrol_robot.launch" />

    <!--     gps_loc -->
<!--     <node pkg="patrol_robot" type="gps_loc.py" name="gps_loc"/> -->

    <!--     fws_simple_move_base_server -->
    <node pkg="patrol_robot" type="fws_simple_move_base_server.py" name="fws_simple_move_base_server" output="screen" />

<!--      --><!--     simple_move_base_client -->
<!--     <node pkg="patrol_robot" type="simple_move_base_client.py" name="simple_move_base_client" output="screen" /> -->

    <!--    task manager, the clients for move base and line track -->
    <node pkg="patrol_robot" type="fws_task_manager.py" name="fws_task_manager" output="screen" />

    <!--    webgui -->
    <node pkg="patrol_robot" type="fws_webgui.py" name="fws_webgui" output="screen" />

    <!--       line_track server -->
    <node pkg="patrol_robot" type="line_track_server.py" name="line_track_server" output="screen" />

    <!--   move_base -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(arg param_yaml_file)" command="load"/>
        <param name="planner_frequency" value="1" />
        <param name="controller_frequency" value="1" />
        <param name="controller_patience" value="15" />
        <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />
        <remap from="/cmd_vel" to="/cmd_vel_from_move_base"/>
    </node>

    <!--   local_planner_pure_pursuit_fws -->
    <node pkg="patrol_robot" type="local_planner_pure_pursuit_fws.py" name="local_planner_pure_pursuit_fws" output="screen" />

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find patrol_robot)/rviz_cfg/fws_sim.rviz"/>
</launch> 

